crates/matrix-sdk-ui/src/timeline/inner/mod.rs

```diff
@@ -718,9 +719,11 @@ impl<P: RoomDataProvider> TimelineInner<P> {
         txn.items.set(idx, new_item);
 
         if is_error {
-            // When there is an error, sending further messages is paused. This
-            // should be reflected in the timeline, so we set all other pending
-            // events to cancelled.
+            // When there is an error, sending further messages is paused. This should be
+            // reflected in the timeline, so we set all other pending events to
+            // cancelled.
+            //
+            // TODO(bnjbvr): spooky action at a distance here^
```

Comment:
While I like the physics reference and don't want you to remove it, maybe it would make sense what the physicist in you meant here? I can guess that it's probably about one item affecting/cancelling other items, it would be nice to write it down.

Reply:
Hah fun, I for one didn't know it was a physics reference xD

https://en.wikipedia.org/wiki/Action_at_a_distance_(computer_programming)

In this case, there's tight coupling between this loop and the behavior of the sending queue, which is just working by coincidence (both paths know they interact together that way), and it's not good. Will update the comment.